core/src/main/java/com/linecorp/armeria/common/metric/DefaultMeterIdPrefixFunction.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.metric;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+
+import com.linecorp.armeria.common.Flags;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.common.logging.RequestLogProperty;
+import com.linecorp.armeria.common.logging.RequestOnlyLog;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+
+/**
+ * Default {@link MeterIdPrefixFunction} implementation.
+ */
+public class DefaultMeterIdPrefixFunction implements MeterIdPrefixFunction {
```

Comment:
Could we hide this class from the public API and allow a user specify some functions when calling a static factory method in `MeterIdPrefixFunction`? `GrpcMeterIdPrefixFunction` could then become a simple shortcut to that method. (composition over inheritance)